### ISSUE TITLE
Bug: [no-base-to-string] Custom classes extending from Error errors wrongly

### ISSUE THREAD

**Issue by TheMythologist** [typescript-eslint/typescript-eslint#7385]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.1.6&fileType=.ts&code=MYGwhgzhAEDCCuEAuB7AtgUQE5ZV6ApgB5IEB2AJjNrvgN7QC%2BAsAFBvApnKE57QBeaGQIB3OIlSY%2BWABQByYJPS9a0ABYEsBeQEo2HLj2RYAlmQDmg1XgB0qAMpIzl2bqA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oCMBDZRLXxdk%2BKkwDm6KImjQO0SODABfECqA&tsconfig=N4XyA&tokens=false

### Repro Code

```typescript
class CustomError extends Error {}

const error = new CustomError('custom error here')

const string = error.toString()
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-base-to-string": "error"
  }
}
```

### Expected Result

The rule `no-base-to-string` should not error as `Error` classes implement custom `toString` methods.

### Actual Result

```js
'error' will evaluate to '[object Object]' when stringified. 5:16 - 5:21
````

### Additional Info

I'm not sure if this is exactly a `typescript-eslint` issue or just a `typescript` issue. This is not reproducible when a class extends from `URLSearchParams` or `URL`.

**Comment by bradzacher** [comment]
https://typescript-eslint.io/rules/no-base-to-string/#ignoredtypenames

This is working as expected - `Error` doesn't have `toString` defined on it - so `CustomError` doesn't have `toString` defined on it - so the rule errors.

We can only work based on the types.

**Comment by TheMythologist** [comment]
Should I raise an issue to Typescript instead, that the `toString` function should be declared on `Error`?

**Comment by TheMythologist** [comment]
Also as an additional feature, should we ignore classes that extend from classes defined in `ignoredTypeNames`?

**Comment by bradzacher** [comment]
> Should I raise an issue to Typescript instead, that the toString function should be declared on Error?

Not likely to change any time soon... https://github.com/microsoft/TypeScript/issues/38347
